src/goto-instrument/cover_basic_blocks.cpp

```diff
@@ -58,13 +58,7 @@ cover_basic_blockst::cover_basic_blockst(const goto_programt &goto_program)
 
     block_map[it] = current_block;
 
-    // update lines belonging to block
-    const irep_idt &line = it->source_location.get_line();
-    if(!line.empty())
-    {
-      block_info.lines.insert(unsafe_string2unsigned(id2string(line)));
-      block_info.source_lines.insert(it->source_location);
-    }
+    add_block_lines(block_info, *it);
```

Comment:
This is really unfortunate. Constructors generally shouldn’t be as complicated as this one is to begin with!